cidr/cidr.go

```diff
@@ -110,3 +110,91 @@ func AddressCount(network *net.IPNet) uint64 {
 	prefixLen, bits := network.Mask.Size()
 	return 1 << (uint64(bits) - uint64(prefixLen))
 }
+
+//VerifyNetwork takes a list subnets and supernet (CIDRBlock) and verifies
+//none of the subnets overlap and all subnets are in the supernet
+//it returns an error if any of those conditions are not satisfied
+func VerifyNetwork(subnets []*net.IPNet, CIDRBlock *net.IPNet) error {
```

Comment:
I like the idea of this function, but I think the name doesn't really explain what it does well.

What do you think of `VerifyNonOverlapping`, or `VerifyNoOverlap`?